src/charm.py

```diff
@@ -1142,26 +1159,62 @@ def _plugin_swift_reconciliation(self):
             "serve-from-swift",
             "remove-local-file",
         ]
-        enable_swift = bool(swift_config_str.strip())
-        if not enable_swift:
+        swift_config = safe_load(swift_config_str)
+        if not swift_config:
+            return {}
+        if "url" in swift_config:
+            swift_url = swift_config["url"]
+            swift_url = re.sub("/wp-content/uploads/?$", "", swift_url)
+            swift_url = swift_url[: -(1 + len(swift_config.get("bucket", "")))]
```

Comment:
Probably a comment would be interesting here as this is not trivial logic

Reply:
Add some comments, thanks!